sqlmesh/core/engine_adapter/snowflake.py

```diff
@@ -99,6 +102,97 @@ def snowpark(self) -> t.Optional[SnowparkSession]:
             ).getOrCreate()
         return None
 
+    def create_managed_table(
+        self,
+        table_name: TableName,
+        query: Query,
+        columns_to_types: t.Optional[t.Dict[str, exp.DataType]] = None,
+        partitioned_by: t.Optional[t.List[exp.Expression]] = None,
+        clustered_by: t.Optional[t.List[str]] = None,
+        table_properties: t.Optional[t.Dict[str, exp.Expression]] = None,
+        table_description: t.Optional[str] = None,
+        column_descriptions: t.Optional[t.Dict[str, str]] = None,
+        **kwargs: t.Any,
+    ) -> None:
+        target_table = exp.to_table(table_name)
+
+        # Snowflake defaults to uppercase and it also makes the property presence checks
+        # easier
+        table_properties = {k.upper(): v for k, v in (table_properties or {}).items()}
+
+        # the WAREHOUSE property is required for a Dynamic Table
+        if "WAREHOUSE" not in table_properties:
+            table_properties["WAREHOUSE"] = self._current_warehouse
+
+        # so is TARGET_LAG
+        # ref: https://docs.snowflake.com/en/sql-reference/sql/create-dynamic-table
+        if "TARGET_LAG" not in table_properties:
+            raise SQLMeshError(
+                "`target_lag` must be specified in the model physical_properties for a Snowflake Dynamic Table"
+            )
+
+        source_queries, columns_to_types = self._get_source_queries_and_columns_to_types(
+            query, columns_to_types, target_table=target_table
+        )
+
+        self._create_table_from_source_queries(
+            quote_identifiers(target_table),
```

Comment:
Why are we quoting here? Note that by default `_to_sql` quotes the identifiers in the expression.

Reply:
I think I saw it used elsewhere in a similar situation. Monkey see, monkey do :)

But you're right, it appears to be unnecessary. I've removed it